### ISSUE TITLE
Unexpected exception in last cell of the first tutorial

### ISSUE THREAD

**Issue by rgerkin** [brian-team/brian2#1114]
In the first tutorial (`brian2/tutorials/1-intro-to-brian-neurons.ipynb`) on the master branch (cloned a few minutes ago), I get the following exception for the last cell (unexpected; I recognize that some of the exceptions raised in earlier cells are by design):

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-26-d3a3c36b1c36> in <module>
     29 run(duration)
     30 
---> 31 _ = hist(spikemon.t/ms, 100, histtype='stepfilled', facecolor='k', weights=ones(len(spikemon))/(N*defaultclock.dt))
     32 xlabel('Time (ms)')
     33 ylabel('Instantaneous firing rate (sp/s)');

/opt/conda/lib/python3.7/site-packages/matplotlib/pyplot.py in hist(x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked, normed, data, **kwargs)
   2634         align=align, orientation=orientation, rwidth=rwidth, log=log,
   2635         color=color, label=label, stacked=stacked, normed=normed,
-> 2636         **({"data": data} if data is not None else {}), **kwargs)
   2637 
   2638 

/opt/conda/lib/python3.7/site-packages/matplotlib/__init__.py in inner(ax, data, *args, **kwargs)
   1587     def inner(ax, *args, data=None, **kwargs):
   1588         if data is None:
-> 1589             return func(ax, *map(sanitize_sequence, args), **kwargs)
   1590 
   1591         bound = new_sig.bind(ax, *args, **kwargs)

/opt/conda/lib/python3.7/site-packages/matplotlib/axes/_axes.py in hist(self, x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked, normed, **kwargs)
   6690 
   6691         if len(w) != nx:
-> 6692             raise ValueError('weights should have the same shape as x')
   6693 
   6694         for xi, wi in zip(x, w):

ValueError: weights should have the same shape as x
```

**Comment by Vigneswaran-Chandrasekaran** [comment]
Hi, I am newbie to Brian. As each cell in the notebook contains `start_scope()` line in the top, I guess there wouldn't be any previous cells's influence, I'm afraid.

**Comment by rgerkin** [comment]
No, I suspect it is an incompatibility between the version of Brian2 against which this notebook was written and the version currently in the master branch.

**Comment by mstimberg** [comment]
Hi, thanks for your bug report. I cannot reproduce this problem on my machine, latest master (and Python 3.7 + matplotlib 3.1.1) works fine for that cell. I have a hard time seeing how the shapes could be different here, what does
```Python
print((spikemon.t/ms).shape)
print((ones(len(spikemon))/(N*defaultclock.dt)).shape)
```
give you?

**Comment by rgerkin** [comment]
@mstimberg It turns out to be the difference between Matplotlib 3.1.0 and 3.1.1, in particular [this commit](https://github.com/matplotlib/matplotlib/commit/9eac832db3b5e281b98d679577fbcff8e93d109f#diff-6a88bd9ec92af89e699c0cf8ec8ffa28) which didn't get merged until 3.1.1.  

Basically, Matplotlib reshapes both the x's and the weights before plotting, but in 3.1.0 it does not successfully reshape them in the same way, and 3.1.1 fixed this bug.  

The tutorial can be hacked to run in either 3.1.0 or 3.1.1 by, for example, changing:
```
_ = hist(spikemon.t/ms, 100, histtype='stepfilled', facecolor='k', weights=ones(len(spikemon))/(N*defaultclock.dt))
```
to
```
_ = hist(spikemon.t/ms, 100, histtype='stepfilled', facecolor='k', weights=list(ones(len(spikemon))/(N*defaultclock.dt)))
```

I don't see matplotlib in your `setup.py` file, since it isn't required to actually run simulations in Brian, so I've submitted PR #1115 for the tutorial to ensure it runs for people using Matplotlib <= 3.1.0.  Although presumably the tutorial worked 2 years ago upon the previous commit, for whatever version of Matplotlib you used then.  So imagine this was a temporary regression from some Matplotlib 2.x through 3.1.0.

**Comment by mstimberg** [comment]
Closed via #1115.
